File: BAEKJOON/src/bronze/ii/java/BOJ_2798_Blackjack.java
package bronze.ii.java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

// 블랙잭
public class BOJ_2798_Blackjack {

    private static int blackjack(int n, int m, int[] arr) {

        int sum;
        int res = 0;
        for (int i = 0; i < n - 2; i++) {
            for (int j = i + 1; j < n - 1; j++) {
                for (int k = j + 1; k < n; k++) {
                    sum = arr[i] + arr[j] + arr[k];
                    if (sum <= m) res = Math.max(res, sum);
                    else break;
                }
            }
        }
        return res;
    }

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        final int N = Integer.parseInt(st.nextToken());
        final int M = Integer.parseInt(st.nextToken());
//        int[] cards = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).sorted().toArray();
        st = new StringTokenizer(br.readLine());
        int[] cards = new int[N];
        for (int i = 0; i < N; i++) {
            cards[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(cards);
        System.out.print(blackjack(N, M, cards));
    }
}

File: BAEKJOON/src/bronze/ii/py/BOJ_2798_Blackjack.py
# 블랙잭
import sys


def blackjack(n, m, arr):
    res = 0
    for i in range(n - 2):
        for j in range(i + 1, n - 1):
            for k in range(j + 1, n):
                tot = arr[i] + arr[j] + arr[k]
                if tot <= m:
                    res = max(res, tot)
                else:
                    break
    return res


N, M = map(int, sys.stdin.readline().split())
cards = list(map(int, sys.stdin.readline().split()))
cards.sort()
print(blackjack(N, M, cards))
